packages/teams-js/src/public/app.ts

```diff
@@ -806,6 +816,100 @@ export namespace app {
       resolve(sendAndHandleStatusAndReason('executeDeepLink', deepLink));
     });
   }
+
+  /**
+   * Namespace to opt into full-caching or delayed-termination of an app.
+   *
+   * @beta
+   */
+  export namespace lifecycle {
+    /**
+     * Register on resume handler function type
+     *
+     * @param context - Data structure to be used to pass the context to the app.
+     */
+    type registerOnResumeHandlerFunctionType = (context: ResumeContext) => void;
+    /**
+     * A function that is called when the app is ready to be suspended or terminated.
+     */
+    type ReadyToSuspendOrTerminateFunction = () => void;
+
+    /**
+     * Register before suspendOrTerminate handler function type
+     *
+     * @param readyToSuspendOrTerminate - A function that the handler is called with. Apps should call this function when they are ready.
+     * @returns void
+     */
+    type registerBeforeSuspendOrTerminateHandlerFunctionType = (
+      readyToSuspendOrTerminate: ReadyToSuspendOrTerminateFunction,
+    ) => void;
+
+    /**
+     * Registers a handler to be called before the page is suspended or terminatd.
+     *
+     * @param handler - The handler to invoke before the page is suspended or terminated. The page should
+     * invoke the readyToSuspendOrTerminate function provided to it once it's ready.
+     *
+     */
+    export function registerBeforeSuspendOrTerminateHandler(
+      handler: registerBeforeSuspendOrTerminateHandlerFunctionType,
+    ): void {
+      handler && ensureInitialized(runtime);
+      if (handler && !isSupported()) {
+        throw errorNotSupportedOnPlatform;
+      }
+      Handlers.registerBeforeSuspendOrTerminateHandler(handler);
+    }
+
+    /**
+     * Checks if app.lifecycle is supported by the host.
+     * @returns boolean to represent whether the lifecycle capability is supported
+     * @throws Error if {@linkcode app.initialize} has not successfully completed
+     *
+     * @beta
+     */
+    export function isSupported(): boolean {
+      return ensureInitialized(runtime) && runtime.supports.app && runtime.supports.app.lifecycle ? true : false;
+    }
+
+    /**
+     * Namespace to opt into full caching of an app
+     *
+     * @beta
+     */
+    export namespace caching {
+      /**
+       * Registers a handler to be called when the page has been requested to load.
```

Comment:
>load [](http://example.com/codeflow?start=78&length=4)

Is it worth (and would it be correct) to expand on this to say "requested to resume from being suspended?"

Clearly there is a longer narrative needed to explain the app lifecycle, and I'm sure that we'll work with Erika to document and publish that, and then link to it from the code. So I'm not suggesting that the full flow needs to be documented here. But maybe a few details to improve the summary docs might help?

Reply:
Yep talking to Erica about it is on my list!